contracts/Tokens/VAI/VAIController.sol

```diff
@@ -409,6 +448,16 @@ contract VAIController is VAIControllerStorageG2, VAIControllerErrorReporter, Ex
 
     // solhint-disable-next-line code-complexity
     function getMintableVAI(address minter) public view returns (uint, uint) {
+        if (mintEnabledOnlyForPrimeHolder && prime == address(0)) {
+            return (uint(Error.REJECTION), 0);
+        }
```

Comment:
I would move this check to the `toggleOnlyPrimeHolderMint` function, to save gas in the user operations. If the new value for `mintEnabledOnlyForPrimeHolder` is true and `prime == address(0)`, I would revert the TX